tests/components/minecraft_server/test_config_flow.py

```diff
@@ -30,12 +35,14 @@ async def test_show_config_form(hass: HomeAssistant) -> None:
     assert result["step_id"] == "user"
 
 
-async def test_address_validation_failed(hass: HomeAssistant) -> None:
+async def test_address_validation_failure(hass: HomeAssistant) -> None:
     """Test error in case of a failed connection."""
     with patch(
-        "homeassistant.components.minecraft_server.api.MinecraftServer.__init__",
-        side_effect=[MinecraftServerAddressError, MinecraftServerAddressError],
-        return_value=None,
+        "mcstatus.server.BedrockServer.lookup",
```

Comment:
Does it work to patch 
`homeassistant.components.minecraft_server.api.BedrockServer.lookup` to limit the scope of the patch?

Guidance from https://docs.python.org/3/library/unittest.mock.html#where-to-patch 
> The basic principle is that you patch where an object is *looked up*, which is not necessarily the same place as where it is defined.

Reply:
It looks like it's working.

@MartinHjelmare Since you gave me this review comment [here](https://github.com/home-assistant/core/pull/100925#discussion_r1351810032), can you please tell us what your favorite solution would be? Thanks!

Reply:
> It looks like it's working.

To clarify, I'm not saying it doesn't work.

> 
> @MartinHjelmare Since you gave me this review comment [here](https://github.com/home-assistant/core/pull/100925#discussion_r1351810032), can you please tell us what your favorite solution would be? Thanks!

To clarify, i'm not disagreeing. We should still patch the library code, not our code. These aren't mutually exclusive comments.

Reply:
> To clarify, I'm not saying it doesn't work.
> 
> > I meant that I tested your proposal and that it works fine. With your proposal we still patch the library, just at another place.

Reply:
I normally try to patch the library in the `homeassistant` path where it is used. That always works. Patching the library in the library path sometimes works and sometimes won't work.

Reply:
Done, thank you both 👍